pwndbg/gdblib/disasm/__init__.py

```diff
@@ -416,7 +416,7 @@ def near(
         while insn is not None and len(insns) < instructions:
             if DEBUG_ENHANCEMENT:
                 print(f"Got instruction from cache, addr={cached:#x}")
-            if insn.jump_like and insn.split == SplitType.NO_SPLIT:
+            if insn.jump_like and insn.split == SplitType.NO_SPLIT and not insn.causes_branch_delay:
```

Comment:
Nitpick: https://docs.python.org/3/howto/enum.html#comparisons

We should probably compare enumerations via `is` instead of `==`? Unless `insn.split` is just an int. If it is an int, then `is` will work here due to small numbers being singletons in Python, but overall `x is y` for `x,  y: int` is not correct.